src/main/core/controller.c

```diff
@@ -224,14 +219,14 @@ typedef struct _ProcessCallbackArgs {
     const char* hostname;
 } ProcessCallbackArgs;
 
-static void _controller_registerProcessCallback(const ProcessOptions* proc, void* _callbackArgs) {
+static int _controller_registerProcessCallback(const ProcessOptions* proc, void* _callbackArgs) {
     ProcessCallbackArgs* callbackArgs = _callbackArgs;
 
     char* plugin = processoptions_getPath(proc);
     if (plugin == NULL) {
         error("For host '%s', couldn't find program path: '%s'", callbackArgs->hostname,
               processoptions_getRawPath(proc));
-        exit(EXIT_FAILURE);
+        return -1;
```

Comment:
I mainly changed this to keep it consistent with the surrounding code, which also bubbles the error up. Would it be better to replace the other code which bubbles the error with exit() calls instead?

Reply:
I'm ok with either of exit-immediately or bubble up with warn_unused_result. Latter is probably a little better since it lets us log any other relevant context as we bubble back up the stack.

Reply:
Okay I left it as bubble-up with a warn_unused_result.